### DISCUSSION TITLE
New mod select menu side scrolling with mouse is inverted

### DISCUSSION THREAD

**Discussion by M0rica** [ppy/osu#18175] [Q&A] [answered]
### Type

Cosmetic

### Bug description

Side scrolling with mouse wheel in the new mod select menu is inverted, when I scroll left the menu scrolls right and vice versa.

### Screenshots or videos

https://user-images.githubusercontent.com/65033374/167388825-4d1f659b-abdc-4d46-9880-f0ea2528c25a.mp4

### Version

2022.509.0

### Logs

I don't think this needs logs.

**Comment by frenzibyte** [comment]
I suppose you're holding <kbd>Shift</kbd> while scrolling here?

**Comment by peppy** [comment]
This is probably a system setting. Can't help because you provided nothing.

**Comment by SonoSooS** [comment]
I'm also having this issue.

It turns out that "natural scrolling" (in Windows it has no name) inverts both X and Y scrolling (checked the Window Messages sent to the osu! window).

Turning off natural scrolling results in the mod list scrolling horizontally as expected with a touchpad as if natural scrolling was enabled, but now scrolling a mod column is in the wrong direction, meaning that the mod list horizontal scrolling direction is indeed inverted.

Edit: to be clear, only horizontal scrolling to scroll horizontally is inverted. Touchpad users with natural scrolling enabled should not be scrolling vertically to move horizontally anyway, as it'll feel weird and inverted, and that behavior works correctly for a mouse user.

Logs:
[runtime.log](https://github.com/ppy/osu/files/8651159/runtime.log)
[performance.log](https://github.com/ppy/osu/files/8651160/performance.log)
[updater.log](https://github.com/ppy/osu/files/8651161/updater.log)
[database.log](https://github.com/ppy/osu/files/8651162/database.log)

**Comment by cdwcgt** [comment]
I also have a problem with side scrolling. But my problem is I can't move like in the video using mouse wheel even hold `shift`
It awful for me because I can only scroll through the slider below
[database.log](https://github.com/ppy/osu/files/8651673/database.log)
[performance.log](https://github.com/ppy/osu/files/8651674/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8651675/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8651676/updater.log)

**Reply by frenzibyte** [reply]
You can drag through sections by mouse (holding from one position and dragging to another position), or scroll while hovering on a mod section's header.

**Comment by peppy** [comment, marked as answer]
This is going to be a windows-specific issues with scroll handling, and will need to be fixed at the framework side. It works correctly on macOS with precision / natural scroll devices, and also on mobile devices with tap drag.

Tracking via https://github.com/ppy/osu-framework/issues/5167.
